lib/fluent/plugin/in_system_metrics.rb

```
@@ -0,0 +1,143 @@
+        metrics
+      end
+
+      def collect_memory_metrics
+        metrics = {}
+        begin
+          if File.exist?('/proc/meminfo')
+            # Linux memory metrics
+            mem_info = File.read('/proc/meminfo')
+            total = mem_info.match(/MemTotal:\s+(\d+)/)[1].to_i * 1024
+            free = mem_info.match(/MemFree:\s+(\d+)/)[1].to_i * 1024
+            available = mem_info.match(/MemAvailable:\s+(\d+)/)[1].to_i * 1024 rescue free
```

RuboCop (1.75.5)

[convention] Avoid using `rescue` in its modifier form.

(Style/RescueModifier)